contracts/mocks/MockValidator.sol

```diff
@@ -28,11 +29,13 @@ contract MockValidator is IValidator {
     }
 
     function onInstall(bytes calldata data) external {
+        require(IModuleManager(msg.sender).isModuleInstalled(MODULE_TYPE_VALIDATOR, address(this), ""), "Validator is still installed");
```

Comment:
I should be able to call onInstall again on already installed module right?
for example K1 validator I might want to swap an owner

Reply:
The `onInstall` function should only be called within an `installModule`. It's a mock, but we need to consider the possibility that `onInstall` could be called directly. We should think about how to secure it or call it properly without causing any issues so I think it's fine here as MockValidator is not meant to be used on prod

Reply:
even if there's a real K1 validator (check one of the modules from module-kit which are being audited) no one would write two different methods onInstall and transferOwnership I feel. if it's directly called it's always referred by msg.sender so it shouldn't be an issue. imo this should be removed. test should check onInstall was called for smart account or some random sender

Reply:
So, we acknowledge this and accept the potential risk of breaking a Smart Account if it's mistakenly used.

Reply:
as discussed on mock validator it should be fine. but we should address it (maybe chat with Rhinestone also?) before we start K1 validator in prod on deployed accounts.

Reply:
maybe this too for tomorrow's call